### PR TITLE
Python: Fix header conversion to byte-pair on scope building

### PR DESCRIPTION
**Pull request by morgan9e**
Reopen of #2125

### CODE REVIEW

**Comment by morgan9e** [comment]
Should I add more tests?

File: src/workerd/server/tests/python/asgi/asgi.wd-test [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        ],
+        bindings = [
+          ( name = "SELF", service = "python-asgi" ),
+        ],
+        compatibilityDate = "2024-10-01",
+        compatibilityFlags = ["python_workers"],
+      )
+    )
+  ],
+);
```
Comment (hoodmane) [inline src/workerd/server/tests/python/asgi/asgi.wd-test:19]: Please add a newline.

**Review by hoodmane — APPROVED** [review]
Thanks for your patience @morgan9e. Looks good to me, just please add a new line.

**Comment by hoodmane** [comment]
Test doesn't seem to pass in CI...

**Comment by morgan9e** [comment]
> Test doesn't seem to pass in CI...

working on it, but it passes when I run it on replicated environment (with same setup script as CI)

**Comment by morgan9e** [comment]
It seems to pass the test

**Comment by hoodmane** [comment]
Thanks for your patience @morgan9e!

**Pull request merged**